docs/src/index.md

```diff
@@ -0,0 +1,68 @@
+# Pkg3.jl
+
+!!! warning
+    This documentation is a work in progress and the information in it might be or become outdated.
+
+Sections:
+
+```@contents
+Pages = [
+    "index.md"]
+```
+
+## Introduction
+
+Pkg3 is the package manager for Julia.
+
+## Getting Started
+
+The Pkg REPL-mode is entered using from the Julia REPL using the key `]`. To start a new project, use `pkg> new`.
+
+```
+pkg> new HelloWorld
+```
+
+This creates a new package `HelloWorld` with the following files
+
+```jl
+julia> cd("HelloWorld")
+shell> tree .
+.
+├── Project.toml
+└── src
+    └── HelloWorld.jl
+
+1 directory, 2 files
+```
+
+The `Project.toml` file contains the name of the package, its unique UUID, its version, the author and eventual dependencies:
+
+```toml
+name = "HelloWorld"
+uuid = "b4cd1eb8-1e24-11e8-3319-93036a3eb9f3"
+version = "0.1.0"
+author = ["Some One <someone@email.com>"]
+
+[deps]
+```
+
+The content of `src/HelloWorld.jl` is:
+
+```jl
+__precompile__(true)
```

Comment:
I feel like this has all the downsides of making `__precompile__(true)` the default without the upsides: people will just write code and if there are precompilation hazards then they will hit them without realizing it. But we still have to have this ugly thing at the top of the package entry point file. If we're going to generate this, we may as well default to having this be on. Ref: https://github.com/JuliaLang/julia/issues/26282.

Reply:
True

Reply:
Sorry for interjecting: wouldn't it be easier for beginners to do the opposite: 1) turn precompilation on be default (I'm generally in favor of this), 2) generate `__precompile__(false)` in a stub, possibly with the comment: "erase if confident" or similar.